### PR TITLE
ISPN-14628 LLEN command

### PR DESCRIPTION
**Pull request by karesti**
https://issues.redhat.com/browse/ISPN-14628

### CODE REVIEW

**Comment by karesti** [comment]
@jabolina rebased from main

**Review by jabolina — APPROVED** [review]
Looks good, needs another rebase.

**Comment by jabolina** [comment]
Integrated into main, thanks @karesti!

File: server/resp/src/main/java/org/infinispan/server/resp/commands/list/LLEN.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      return indexAndReturn(handler, ctx, arguments);
+   }
+
+   protected CompletionStage<RespRequestHandler> indexAndReturn(Resp3Handler handler,
+                                                              ChannelHandlerContext ctx,
+                                                              List<byte[]> arguments) {
+      byte[] key = arguments.get(0);
+      EmbeddedMultimapListCache<byte[], byte[]> listMultimap = handler.getListMultimap();
+
+      return CompletionStages.handleAndCompose(listMultimap.size(key) ,(size, t) -> {
```
Comment (wburns) [inline server/resp/src/main/java/org/infinispan/server/resp/commands/list/LLEN.java:51]: Couldn't this just use `thenApply` on the stage returned from size method?

File: server/resp/src/main/java/org/infinispan/server/resp/commands/list/LLEN.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+   public LLEN() {
+      super(2, 1, 1, 1);
+   }
+
+   @Override
+   public CompletionStage<RespRequestHandler> perform(Resp3Handler handler,
+                                                      ChannelHandlerContext ctx,
+                                                      List<byte[]> arguments) {
+
+      if (arguments.size() < 1) {
```
Comment (wburns) [inline server/resp/src/main/java/org/infinispan/server/resp/commands/list/LLEN.java:36]: Should we throw an error if argument size is > 1 as well?

**Review by wburns — COMMENTED** [review]
Just some minor things

File: server/resp/src/main/java/org/infinispan/server/resp/commands/list/LLEN.java (hunk 2 above)
Comment (jabolina) [reply server/resp/src/main/java/org/infinispan/server/resp/commands/list/LLEN.java:36]: Coming to think of it, we should add something to validate the arguments with the command arity.

Comment (wburns) [reply server/resp/src/main/java/org/infinispan/server/resp/commands/list/LLEN.java:36]: Yeah, that should work

Comment (karesti) [reply server/resp/src/main/java/org/infinispan/server/resp/commands/list/LLEN.java:36]: yes, I agree. I will do a PR with that in mind so we don't need to code this anymore

Comment (karesti) [reply server/resp/src/main/java/org/infinispan/server/resp/commands/list/LLEN.java:36]: https://issues.redhat.com/browse/ISPN-14896

**Pull request merged**